modules/library/metadata/src/main/java/org/geotools/util/GeoToolsUnitFormat.java

```diff
@@ -86,59 +81,9 @@ public static SimpleUnitFormat getInstance() {
      */
     protected static class BaseGT2Format extends DefaultFormat {
 
-        public BaseGT2Format() {}
-
-        @SuppressWarnings("unchecked")
-        protected void initUnits(UnitFormat base) {
-            /**
-             * Labels and alias are only defined on the DEFAULT format instance, so these
-             * definitions are not inherited by subclassing DefaultFormat. Therefore, we need to
-             * clone these definitions in our GT formats
-             */
-            try {
-                HashMap<String, Unit<?>> nameToUnitMap = getNameToUnitMap(base);
-                HashMap<Unit<?>, String> unitToNameMap = getUnitToNameMap(base);
-
-                for (Map.Entry<String, Unit<?>> entry : nameToUnitMap.entrySet()) {
-                    String name = entry.getKey();
-                    Unit<?> unit = entry.getValue();
-
-                    if (unitToNameMap.containsKey(unit) && name.equals(unitToNameMap.get(unit))) {
-                        label(unit, name);
-                        addUnit(unit);
-                    } else {
-                        alias(unit, name);
-                    }
-                }
-            } catch (Throwable t) {
-                throw new RuntimeException(
-                        "Failed to initialize the "
-                                + this.getClass().getCanonicalName()
-                                + " format unit parser with the same values as "
-                                + base.getClass().getCanonicalName()
-                                + "  one",
-                        t);
-            }
-        }
-
-        @SuppressWarnings("unchecked") // reflection in use, cannot be type safe
-        private HashMap<Unit<?>, String> getUnitToNameMap(UnitFormat base)
-                throws NoSuchFieldException, IllegalAccessException {
-            java.lang.reflect.Field unitToNameField =
-                    DefaultFormat.class.getDeclaredField("unitToName");
-            unitToNameField.setAccessible(true);
-
-            return (HashMap<Unit<?>, String>) unitToNameField.get(base);
-        }
-
-        @SuppressWarnings("unchecked") // reflection in use, cannot be type safe
-        private HashMap<String, Unit<?>> getNameToUnitMap(UnitFormat base)
-                throws NoSuchFieldException, IllegalAccessException {
-            java.lang.reflect.Field nameToUnitField =
-                    DefaultFormat.class.getDeclaredField("nameToUnit");
-            nameToUnitField.setAccessible(true);
-
-            return (HashMap<String, Unit<?>>) nameToUnitField.get(base);
+        public BaseGT2Format() {
+            SimpleUnitFormat.initDefaultFormat(this);
+            this.label(NonSI.DEGREE_ANGLE, "rad");
```

Comment:
@keilw, does this look reasonable to you?

Reply:
At first sight it looks OK, was there some patch of Indriya `2.1.2-SNAPSHOT` you rely on, or would it also work with the final `2.1.1` release?

Reply:
[2.1.1 was released before](https://github.com/unitsofmeasurement/indriya/releases) the [required changes were merged](https://github.com/unitsofmeasurement/indriya/pull/317).

Reply:
There are a few issues marked with the [2.1.2](https://github.com/orgs/unitsofmeasurement/projects/5?card_filter_query=milestone%3A2.1.2) milestone, including the bridge for thise one [316](https://github.com/unitsofmeasurement/indriya/issues/316)

Reply:
So 2.1.2 should be fine, when it is released?